internal/internal_task_handlers.go

```diff
@@ -1265,6 +1270,10 @@ func skipDeterministicCheckForEvent(e *historypb.HistoryEvent) bool {
 		return true
 	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
 		return true
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ACCEPTED,
```

Comment:
Can we check `EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ACCEPTED` ? On replay we skip running the validation and skip adding the accepted message to the outbox no?

Reply:
We don't run validation but we do send the message. The outbox gets drained into a side structure for the nondeterminism check so that we don't actually emit the message back to the server (same as commands)